Detections/OracleDatabaseAudit/OracleDBAuditNewIpForUser.yaml

```diff
@@ -0,0 +1,42 @@
+id: 39a0995e-f4a9-4869-a0ae-36d6d9049bfd
+name: OracleDBAudit - User connected to database from new IP
+description: |
+  'Detects when a user connects to database from new IP address.'
+severity: Low
+requiredDataConnectors:
+  - connectorId: OracleDatabaseAudit
+    dataTypes:
+      - Syslog
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+query: |
+  let lbtime_60d = 60d;
+  let lbtime_24h = 24h;
+  OracleDatabaseAuditEvent
+  | where TimeGenerated between (ago(lbtime_60d) .. ago(lbtime_24h))
+  | where isnotempty(SrcIpAddr)
+  | where isnotempty(DstUserName)
+  | summarize knownIPs = make_set(SrcIpAddr) by DstUserName
+  | join (OracleDatabaseAuditEvent
+        | where TimeGenerated > ago(lbtime_24h)
+        | where isnotempty(SrcIpAddr)
+        | where isnotempty(DstUserName)
+        | where tolower(DbAction) == 'connect'
+        ) on DstUserName
+  | where knownIPs !contains SrcIpAddr

```

Comment:
We should not use contains here, use `| where SrcIpAddr !in (knownIps)`

Reply:
"!in" does not work here. Left "contains"
![image](https://user-images.githubusercontent.com/76435946/111642090-7475f180-8806-11eb-9c03-2a9870c3488c.png)

Reply:
Ah, I didn't notice this before, it is because your ```knownIPs``` set has ```by DstUserName```.  If you remove ```by DstUserName```, it should work and will have better performance.